GameOfLife.java

```diff
@@ -63,16 +67,48 @@ public static int[][] read(String fileName) {
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
-		//// Replace the following statement with your code.
-		return null;
+
+		String line = in.readLine();
+		int row = 0;
+		while (line==""){
+			row++;
+			line = in.readLine();
+		}
+
+		while(line!=""){
```

Comment:
There are can be empty lines in the middle. Probably this is the reason for your bug.